prow/plugins/project/project.go

```diff
@@ -0,0 +1,348 @@
+	// Get all projects for all repos
+	var projects []github.Project
+	for _, repo := range repos {
+		repoProjects, err := gc.GetRepoProjects(org, repo.Name)
+		if err != nil {
+			msg = fmt.Sprintf(errListingRepoProjectsMsg, repo.Name)
+			return gc.CreateComment(org, repo.Name, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+		}
+		projects = append(projects, repoProjects...)
+	}
+
+	// Only fetch org projects if we can't find the proposed project / project to clear in the repo projects
+	updateProjectNameToIDMap(projects)
+	var projectID int
+	var ok bool
+	if projectID, ok = projectNameToIDMap[proposedProject]; !ok {
+		// Get all projects for this org
+		orgProjects, err := gc.GetOrgProjects(org)
+		if err != nil {
+			msg = fmt.Sprintf(errListingOrgProjectsMsg, org)
+			return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+		}
+		projects = append(projects, orgProjects...)
+
+		// If still can't find proposed project / project to clear in the list of projects, abort and create a comment
+		updateProjectNameToIDMap(projects)
+		if projectID, ok = projectNameToIDMap[proposedProject]; !ok {
+			slice := make([]string, 0, len(projectNameToIDMap))
+			for k := range projectNameToIDMap {
+				slice = append(slice, fmt.Sprintf("`%s`", k))
+			}
+			sort.Strings(slice)
+
+			msg = fmt.Sprintf(invalidProject, strings.Join(slice, ", "))
+			return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+		}
+	}
+
+	// Clear issue/PR from project if command is to clear
+	if shouldClear {
+		if err := gc.DeleteProjectCard(projectID); err != nil {
+			msg = fmt.Sprintf(errRemovingFromProjectMsg, proposedProject, projectID, org, repo, e.Number)
+			return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+		}
+		msg = fmt.Sprintf(successClearingProjectMsg, proposedProject)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	// Get all columns for proposedProject
+	projectColumns, err := gc.GetProjectColumns(projectID)
+	if err != nil {
+		msg = fmt.Sprintf(errListingColsMsg, proposedProject)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	// If proposedColumnName is not found (or not provided), add to one of the default
+	// columns. If none of the default columns exists, an error will be shown to the user
+	columnFound := false
+	proposedColumnID := 0
+	for _, c := range projectColumns {
+		if c.Name == proposedColumnName {
+			columnFound = true
+			proposedColumnID = c.ID
+			break
+		}
+	}
+	if !columnFound {
+		// If user does not provide a column name, look for the columns
+		// specified in the project config and see if any of them exists on the
+		// proposed project
+		if proposedColumnName == "" {
+			defaultColumn := ""
```

Comment:
Similar comment here, you could just call `getColumnMap` and not need explicit default values